contributors/design-proposals/storage/container-storage-interface-inline-volumes.md

```diff
@@ -0,0 +1,148 @@
+# In-line CSI volumes in Pods
+
+Author: @jsafrane
+
+## Goal
+* Define API and high level design for in-line CSI volumes in Pod
```

Comment:
After reading through the spec once, there may be two different use cases wrapped up into one here.

1. secrety like volumes. per pod volumes.
2. more traditional volume drivers. volume create/delete/attach/etc.

While both need solving, the vast majority of difficult design/coding I think is around use case number 2.

Since use case number one may be much easier to implement, and would garner a bunch of drivers quickly should we consider breaking the spec into 2, to allow the first use case to be implemented faster then the second?

Reply:
+1. but just hoping to see at least part 1 implemented in 1.12. If the design of part 2 would push implementation out to 1.13, then maybe we should separate. If not, then lets keep it all together.

Reply:
looks like we very well may miss the 1.13 window even for part 1 because we're still debating how to solve part 2. Can we please set aside part 2 for now?